### PR TITLE
Add base support for Firefox

### PR DESCRIPTION
**Pull request by alexesprit**
A couple of compatibility patches for Firefox.

### CODE REVIEW

File: connectors/v2/soundcloud-dom-inject.js [hunk 1]
```diff
@@ -17,12 +17,12 @@ window.SC_ATTACHED = window.SC_ATTACHED || false;
 	}
 	/*jslint unparam: true */
 	window.webpackJsonp([], {
-		0: function (e, t, require) {
-			var modules = require.c, moduleid, el;
+		0: function (e, t, _require) {
```
Comment (inverse) [inline connectors/v2/soundcloud-dom-inject.js]: How does this change work?

File: core/background/main.js [hunk 2]
```diff
@@ -80,7 +80,7 @@ require([
 
 			if (!isActiveSession) {
 				isActiveSession = true;
-				GA.send('pageview', '/background-injected?version=' + chrome.app.getDetails().version);
+				GA.send('pageview', '/background-injected?version=' + chrome.runtime.getManifest().version);
```
Comment (inverse) [inline core/background/main.js]: How does Firefox handle this? 

I was reading a bit about: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/runtime/getManifest

and in their examples they use:

```
browser.runtime.getManifest()
```

File: core/background/notifications.js [hunk 3]
```diff
@@ -126,15 +137,15 @@ define([
 				GA.event('notification', 'authenticate', 'open-unavailable');
 
 				// fallback for browsers with no notifications support
-				window.open(authUrl, 'scrobbler-auth');
+				chrome.tabs.create({url: authUrl});
```
Comment (inverse) [inline core/background/notifications.js]: Same with this

**Comment by inverse** [comment]
Just some comments really, haven't tested.

File: core/background/main.js (hunk 2 above)
Comment (alexesprit) [inline core/background/main.js]: Don't know about the example, but `chrome.runtime.getManifest` works fine as well as other Chrome API functuions (those, that implemented in Firefox, of course).

File: connectors/v2/soundcloud-dom-inject.js (hunk 1 above)
Comment (alexesprit) [inline connectors/v2/soundcloud-dom-inject.js]: AMO linter doesn't like variable named as `require`:
```
1) Unexpected global passed as an argument
Passing a global as an argument is not recommended. Please make this a var instead.
connectors/v2/soundcloud-dom-inject.js line 25
```

**Comment by f1u77y** [comment]
What's blocking the merge?

**Pull request merged**